financial-connections/src/main/java/com/stripe/android/financialconnections/presentation/FinancialConnectionsSheetNativeViewModel.kt

```diff
@@ -239,70 +234,66 @@ internal class FinancialConnectionsSheetNativeViewModel @Inject constructor(
      * 3. User closes without an error, and fetching accounts returns NO accounts. That's a cancel.
      */
     private fun closeAuthFlow(
-        earlyTerminationCause: Terminate.EarlyTerminationCause? = null,
+        earlyTerminationCause: EarlyTerminationCause? = null,
         closeAuthFlowError: Throwable? = null
-    ) {
-        viewModelScope.launch {
-            kotlin
-                .runCatching {
-                    completeFinancialConnectionsSession(
-                        terminalError = earlyTerminationCause?.value
+    ) = viewModelScope.launch {
+        mutex.withLock {
+            // prevents multiple complete triggers.
+            if (awaitState().completed) return@launch
+            setState { copy(completed = true) }
```

Comment:
this is the only difference (closing logic remains the same) -> this prevents close to be called more than once.